File: core/src/test/java/me/raatiniemi/worker/domain/interactor/IsProjectActiveTest.java
/*
 * Copyright (C) 2017 Worker Project
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, version 2 of the License.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package me.raatiniemi.worker.domain.interactor;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import me.raatiniemi.worker.domain.exception.DomainException;
import me.raatiniemi.worker.domain.model.TimeInterval;
import me.raatiniemi.worker.domain.repository.TimeIntervalRepository;
import me.raatiniemi.worker.factory.TimeIntervalFactory;
import me.raatiniemi.worker.util.Optional;

import static junit.framework.Assert.assertFalse;
import static junit.framework.Assert.assertTrue;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

@RunWith(JUnit4.class)
public class IsProjectActiveTest {
    private TimeIntervalRepository timeIntervalRepository;

    @Before
    public void setUp() throws Exception {
        timeIntervalRepository = mock(TimeIntervalRepository.class);
    }

    @Test
    public void execute_withoutTime() throws DomainException {
        when(timeIntervalRepository.findActiveByProjectId(1L))
                .thenReturn(Optional.empty());

        IsProjectActive isProjectActive = new IsProjectActive(timeIntervalRepository);
        assertFalse(isProjectActive.execute(1L));
    }

    @Test
    public void execute_withActiveTime() throws DomainException {
        TimeInterval timeInterval = TimeIntervalFactory.builder()
                .stopInMilliseconds(0L)
                .build();
        when(timeIntervalRepository.findActiveByProjectId(1L))
                .thenReturn(Optional.of(timeInterval));

        IsProjectActive isProjectActive = new IsProjectActive(timeIntervalRepository);
        assertTrue(isProjectActive.execute(1L));
    }
}

File: core/src/test/java/me/raatiniemi/worker/domain/interactor/IsProjectActiveTest.kt
/*
 * Copyright (C) 2018 Tobias Raatiniemi
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, version 2 of the License.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package me.raatiniemi.worker.domain.interactor

import me.raatiniemi.worker.domain.model.newTimeInterval
import me.raatiniemi.worker.domain.repository.TimeIntervalInMemoryRepository
import me.raatiniemi.worker.domain.repository.TimeIntervalRepository
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4
import java.util.*

@RunWith(JUnit4::class)
class IsProjectActiveTest {
    private lateinit var repository: TimeIntervalRepository
    private lateinit var isProjectActive: IsProjectActive

    @Before
    fun setUp() {
        repository = TimeIntervalInMemoryRepository()
        isProjectActive = IsProjectActive(repository)
    }

    @Test
    fun `is project active without time interval`() {
        val actual = isProjectActive(1)

        assertFalse(actual)
    }

    @Test
    fun `is project active with active time interval`() {
        val newTimeInterval = newTimeInterval {
            start = Date(1)
            stopInMilliseconds = 0
        }
        repository.add(newTimeInterval)

        val actual = isProjectActive(1)

        assertTrue(actual)
    }
}
